types/model.d.ts

```diff
@@ -74,9 +90,159 @@ export interface ModelProperties {
  * }
  * ```
  */
-export interface ModelSettings extends AnyObject {
+export interface ModelSettings extends AnyObject, ModelUtilsOptions {
   strict?: boolean;
+
+  /**
+   * Set if manual assignment of auto-generated ID values should be blocked.
+   */
   forceId?: boolean;
+
+  /**
+   * @remarks
+   * Defaults to `true`.
+   */
+  idInjection?: boolean;
+  
+  plural?: string;
+
+  http?: {
+    path?: string;
+  }
+
+  /**
+   * @remarks
+   * Alias of {@link ModelSettings.super}. Takes lower precedence.
+   */
+  base?: ModelBaseClass;
+  /**
+   * @remarks
+   * Alias of {@link ModelSettings.base}. Takes lower precedence.
+   */
+  super?: ModelBaseClass;
+  excludeBaseProperties?: string[];
+
+  /**
+   * Indicates if the {@link ModelBaseClass | Model} is attached to the DataS
+   * @internal
+   */
+  unresolved?: boolean;
+
+  indexes?: {
+    [indexJugglerName: string]: {
+      name: string;
+      /**
+       * Comma-separated column names
+       * 
+       * @remarks
+       * Handled by {@link Connector}s directly by default.
+       * 
+       * Overriden by {@link ModelSettings.indexes.keys}.
+       */
+      columns: string;
+
+      /**
+       * Array of column names to create an index.
+       * 
+       * @remarks
+       * Handled by {@link Connector}s directly by default.
```

Comment:
Ditto. What does "by default" mean? What is the circumstance where this is not the case, and what is the impact?